core/src/main/java/org/apache/spark/api/java/function/MapFunction.java

```diff
@@ -23,5 +23,5 @@
  * Base interface for a map function used in Dataset's map function.
  */
 public interface MapFunction<T, U> extends Serializable {
-  U call(T value) throws Exception;
```

Comment:
If they don't, then users must catch all exceptions and can never throw checked exceptions anymore. We shouldn't change this.